packages/front-end/components/SavedQueries/SavedQueriesList.tsx

```diff
@@ -0,0 +1,247 @@
+import { useState, useCallback } from "react";
+import { date, datetime } from "shared/dates";
+import { SavedQuery } from "back-end/src/validators/saved-queries";
+import { useAuth } from "@/services/auth";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import usePermissionsUtil from "@/hooks/usePermissionsUtils";
+import { useSearch } from "@/services/search";
+import MoreMenu from "@/components/Dropdown/MoreMenu";
+import DeleteButton from "@/components/DeleteButton/DeleteButton";
+import Button from "@/components/Button";
+import SqlExplorerModal from "@/components/SchemaBrowser/SqlExplorerModal";
+
+interface Props {
+  savedQueries: SavedQuery[];
+  mutate: () => void;
+}
+
+export default function SavedQueriesList({ savedQueries, mutate }: Props) {
+  const { apiCall } = useAuth();
+  const { getDatasourceById } = useDefinitions();
+  const permissionsUtil = usePermissionsUtil();
+  const [selectedSavedQuery, setSelectedSavedQuery] = useState<
+    SavedQuery | undefined
+  >();
+
+  const {
+    items,
+    searchInputProps,
+    isFiltered,
+    SortableTH,
+    clear,
+    pagination,
+  } = useSearch({
+    items: savedQueries,
+    defaultSortField: "dateUpdated",
+    localStorageKey: "savedqueries",
+    searchFields: ["name^3", "sql"],
+    pageSize: 20,
+  });
+
+  const handleDelete = useCallback(
+    async (query: SavedQuery) => {
+      await apiCall(`/saved-queries/${query.id}`, {
+        method: "DELETE",
+      });
+      mutate();
+    },
+    [apiCall, mutate]
+  );
+
+  const handleEdit = useCallback((query: SavedQuery) => {
+    setSelectedSavedQuery(query);
+  }, []);
+
+  const handleDuplicate = useCallback((query: SavedQuery) => {
+    setSelectedSavedQuery({
+      ...query,
+      id: "",
+      name: `${query.name}-copy`,
+    });
+  }, []);
+
+  const handleRowClick = useCallback(
+    (query: SavedQuery) => {
+      // Only allow row click for queries the user can edit
```

Comment:
question: Do we still want this if the modal is the only way to see the graphs at the moment?